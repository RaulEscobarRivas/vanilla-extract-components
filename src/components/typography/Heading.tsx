import React from "react";
import { useAsProp } from "../../hooks/useAsProp";
import { useHeadingStyles } from "./useHeadingStyles";
import { HeadingProps } from "./types";

const defaultAsAndSize = "h1";

export function Heading({
  color = "gray.800",
  ellipsis,
  size = defaultAsAndSize,
  shadow,
  textAlign,
  ...rest
}: HeadingProps) {
  const [Tag, props] = useAsProp(
    { as: defaultAsAndSize, ...rest },
    {
      className: useHeadingStyles({
        color,
        ellipsis,
        size,
        shadow,
        textAlign,
      }),
    }
  );

  return <Tag {...props} />;
}
