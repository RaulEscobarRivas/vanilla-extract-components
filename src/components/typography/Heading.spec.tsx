import React from "react";
import { render } from "@testing-library/react";
import { Heading } from "./Heading";

describe("Heading", () => {
  it("should render default style", () => {
    const { container } = render(<Heading />);

    expect(container).toMatchSnapshot();
  });

  it("should render custom h3 tag", () => {
    const { container } = render(<Heading as="h3" />);

    expect(container).toMatchSnapshot();
  });

  it("should render size style", () => {
    const { container } = render(<Heading size="h6" />);

    expect(container).toMatchSnapshot();
  });

  it("should render shadow style", () => {
    const { container } = render(<Heading shadow />);

    expect(container).toMatchSnapshot();
  });

  it("should render ellipsis style", () => {
    const { container } = render(<Heading ellipsis />);

    expect(container).toMatchSnapshot();
  });

  it("should render textAlign=left style", () => {
    const { container } = render(<Heading textAlign="left" />);

    expect(container).toMatchSnapshot();
  });

  it("should render textAlign=right style", () => {
    const { container } = render(<Heading textAlign="right" />);

    expect(container).toMatchSnapshot();
  });

  it("should render textAlign=center style", () => {
    const { container } = render(<Heading textAlign="center" />);

    expect(container).toMatchSnapshot();
  });

  it("should render color style", () => {
    const { container } = render(<Heading color="red.600" />);

    expect(container).toMatchSnapshot();
  });

  it("should render variant styles", () => {
    const { container } = render(
      <Heading
        as="h2"
        color="gray.200"
        shadow
        ellipsis
        textAlign="left"
        size="superhero"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
