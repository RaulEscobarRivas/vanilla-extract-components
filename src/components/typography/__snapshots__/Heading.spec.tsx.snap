// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render color style 1`] = `
<div>
  <h1
    class="variants_colorStyle_red.600__nc0lsl9 Heading_sizeStyle_h1__1lih4p21 Heading_headingBase__1lih4p27"
  />
</div>
`;

exports[`Heading should render custom h3 tag 1`] = `
<div>
  <h3
    class="variants_colorStyle_gray.800__nc0lsl4 Heading_sizeStyle_h1__1lih4p21 Heading_headingBase__1lih4p27"
  />
</div>
`;

exports[`Heading should render default style 1`] = `
<div>
  <h1
    class="variants_colorStyle_gray.800__nc0lsl4 Heading_sizeStyle_h1__1lih4p21 Heading_headingBase__1lih4p27"
  />
</div>
`;

exports[`Heading should render ellipsis style 1`] = `
<div>
  <h1
    class="variants_colorStyle_gray.800__nc0lsl4 Heading_sizeStyle_h1__1lih4p21 variants_ellipsisStyle__nc0lsle Heading_headingBase__1lih4p27"
  />
</div>
`;

exports[`Heading should render shadow style 1`] = `
<div>
  <h1
    class="variants_colorStyle_gray.800__nc0lsl4 Heading_sizeStyle_h1__1lih4p21 variants_shadowStyle__nc0lsld Heading_headingBase__1lih4p27"
  />
</div>
`;

exports[`Heading should render size style 1`] = `
<div>
  <h1
    class="variants_colorStyle_gray.800__nc0lsl4 Heading_sizeStyle_h6__1lih4p26 Heading_subheadingBase__1lih4p28"
  />
</div>
`;

exports[`Heading should render textAlign=center style 1`] = `
<div>
  <h1
    class="variants_colorStyle_gray.800__nc0lsl4 variants_textAlignStyle_center__nc0lslj Heading_sizeStyle_h1__1lih4p21 Heading_headingBase__1lih4p27"
  />
</div>
`;

exports[`Heading should render textAlign=left style 1`] = `
<div>
  <h1
    class="variants_colorStyle_gray.800__nc0lsl4 variants_textAlignStyle_left__nc0lslh Heading_sizeStyle_h1__1lih4p21 Heading_headingBase__1lih4p27"
  />
</div>
`;

exports[`Heading should render textAlign=right style 1`] = `
<div>
  <h1
    class="variants_colorStyle_gray.800__nc0lsl4 variants_textAlignStyle_right__nc0lsli Heading_sizeStyle_h1__1lih4p21 Heading_headingBase__1lih4p27"
  />
</div>
`;

exports[`Heading should render variant styles 1`] = `
<div>
  <h2
    class="variants_colorStyle_gray.200__nc0lsl0 variants_textAlignStyle_left__nc0lslh Heading_sizeStyle_superhero__1lih4p20 variants_shadowStyle__nc0lsld variants_ellipsisStyle__nc0lsle Heading_headingBase__1lih4p27"
  />
</div>
`;
